easybuild/easyblocks/g/gcc.py

```diff
@@ -368,15 +413,40 @@ def configure_step(self):
                     'val': os.getenv('LD_LIBRARY_PATH')
                 }
                 env.setvar('LD_LIBRARY_PATH', ld_lib_path)
-                extra_source = {1: "nvptx-tools", 2: "newlib"}[self.iter_idx]
-                extra_source_dirs = glob.glob(os.path.join(self.builddir, '%s-*' % extra_source))
-                if len(extra_source_dirs) != 1:
-                    raise EasyBuildError("Failed to isolate %s source dir" % extra_source)
-                if self.iter_idx == 1:
-                    # compile nvptx-tools
+                if self.build_stages[self.iter_idx] == NVPTX_TOOLS:
+                    # Configure NVPTX tools and build
+                    extra_source_dirs = glob.glob(os.path.join(self.builddir, 'nvptx-tools-*'))
+                    if not extra_source_dirs:
+                        raise EasyBuildError("Failed to isolate NVPTX source dir")
                     change_dir(extra_source_dirs[0])
-                else:  # self.iter_idx == 2
+                    return super(EB_GCC, self).configure_step()
+                elif self.build_stages[self.iter_idx] == AMD_LLVM:
+                    # Setup symlink from LLD source to 'builddir/lld' so that LLVM can find it
+                    lld_source_dirs = glob.glob(os.path.join(self.builddir, 'lld-*'))
+                    if not lld_source_dirs:
+                        raise EasyBuildError("Failed to isolate LLD source dir")
+                    symlink(lld_source_dirs[0], '%s/lld' % self.builddir)
+                    # Build LLD tools from LLVM needed for AMD offloading
+                    extra_source_dirs = glob.glob(os.path.join(self.builddir, 'llvm-*'))
+                    if not extra_source_dirs:
+                        raise EasyBuildError("Failed to isolate LLVM source dir")
```

Comment:
This should never happen after the check in `prepare_step`, just use `self.llvm_subdir` here